src/System.Private.ServiceModel/src/System/ServiceModel/Channels/BufferedWriteStream.cs

```diff
@@ -57,12 +60,12 @@ public BufferedWriteStream(Stream stream, int bufferSize)
         private void EnsureNotClosed()
         {
             if (_stream == null)
-                throw new ObjectDisposedException("BufferedWriteStream");
+                throw new ObjectDisposedException(nameof(BufferedWriteStream));
         }
 
         private void EnsureCanWrite()
         {
-            Contract.Requires(_stream != null);
+            Contract.Assert(_stream != null);
             if (!_stream.CanWrite)
                 throw new NotSupportedException("write");
```

Comment:
I would prefer to change this to a Contract.Assert so as not to run the code on release builds. The original code was general use and any stream could be passed in, but this implementation is only used by WCF code so the stream isn't arbitrary and a debug assert should be sufficient.